Add recipe for org-repeat-by-cron

TomoeMami opened the pull request:
### Brief summary of what the package does

An Org mode task repeater based on Cron expressions. 

Modified from https://github.com/Raemi/org-reschedule-by-rule (MIT).

### Direct link to the package repository

https://github.com/TomoeMami/org-repeat-by-cron.el

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass -

⸺ `org-repeat-by-cron.el` with byte-compile using Emacs 30.1:
```
org-repeat-by-cron.el:265:36: Warning: value from call to `nreverse' is unused
In end of data:
org-repeat-by-cron.el:530:16: Warning: the function `org-todo' might not be defined at runtime.
org-repeat-by-cron.el:528:16: Warning: the function `org-entry-put' might not be defined at runtime.
org-repeat-by-cron.el:511:33: Warning: the function `org-time-string-to-time' might not be defined at runtime.
org-repeat-by-cron.el:508:96: Warning: the function `org-get-scheduled-time' might not be defined at runtime.
org-repeat-by-cron.el:508:64: Warning: the function `org-get-deadline-time' might not be defined at runtime.
org-repeat-by-cron.el:506:76: Warning: the function `org-schedule' might not be defined at runtime.
org-repeat-by-cron.el:506:61: Warning: the function `org-deadline' might not be defined at runtime.
org-repeat-by-cron.el:502:30: Warning: the function `org-entry-get' might not be defined at runtime.
org-repeat-by-cron.el:501:8: Warning: the function `org-back-to-heading' might not be defined at runtime.
org-repeat-by-cron.el:499:10: Warning: the function `org-entry-is-done-p' might not be defined at runtime.
org-repeat-by-cron.el:315:12: Warning: the function `cl-some' might not be defined at runtime.
org-repeat-by-cron.el:273:14: Warning: the function `cl-delete-duplicates' might not be defined at runtime.
```

⸺ `org-repeat-by-cron.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- `;;; Commentary` is much wider than 80 characters
- org-repeat-by-cron.el#L535: Loading a package should rarely add hooks
```
Although I can see why you'd add the hook, it's often better to let users do this in their config (and/or implement it as a globalized minor mode)

TomoeMami commented:
Thank you for your prompt response! I have made the following changes:

- Replaced `(nreverse result)` with `(setq result (nreverse result))`  
- Changed `(eval-when-compile (require 'org))` and `(eval-when-compile (require 'cl-lib))` to `(require 'org)` and `(require 'cl-lib)` respectively  
- Added line breaks for the `;;; Commentary` section  
- Replaced the original `add-hook` with a minor mode `global-org-repeat-by-cron-mode`  
- Added `defgroup org-repeat-by-cron` and converted previous `defvar` declarations to `defcustom`

https://github.com/TomoeMami/org-repeat-by-cron.el/commit/902c1ff1b115d6c47a4494c214d2ed1e3f5cdff6

TomoeMami commented:
Thanks for mention it! `autoload` Added.
https://github.com/TomoeMami/org-repeat-by-cron.el/commit/d3552759098d247c4bc8ecdbdb07c5637354e048

riscy commented:
Thank you!  And welcome to MELPA. :)
